Adds the initial structure for the teia dao token claim page

jagracar opened the pull request.

jagracar committed:
Adds the initial structure for the teia dao token claim page

jagracar commented:
Hi Mel an others. I had to comment the pre-commit code to be able to commit my changes...

This is just the structure to be filled for the token claim page. Feel free to edit and improve it. The utils.js file contains the code (commented) to do the claim button work. It needs some context info, so it probably needs to be put somewhere else

jagracar committed:
fixes mistake when commenting code

jagracar committed:
Update pre-commit

jagracar commented:
I uncommented the pre-commit part using the github UI and it worked XD

melMass commented:
Small tip you can use "--no-verify" to bypass hooks, if you can post the log of why it did not allow it it could be fixed I think!

Zir0h commented on src/pages/dao/utils.js:
```diff
@@ -0,0 +1,95 @@
+    6: "QmbZsvCDiGv4xWeLtMf1wPxjogpvLjaKLwdZe5xQezUWUS",
+    7: "QmbCvvyP2s6nqq1cCW9H617jRGEYoZkV3SoJHCS9sJmsNU",
+    8: "QmRhr7m6zBtU1YFNcLMo34jQVUFdamG3BCGCFjA9AwBEnK"
+};
```
I pinned these so that they can be fetched with cache.teia.rocks

jagracar replied:
Those are from the test we did one month ago. We will need to create new ones with the final distribution

melMass commented:
Very clean!! 👏 
What is the next step @jagracar ?

jagracar commented:
We need to put the js code in utils.js somewhere else where it can access the context. Also add all the text explanation about the token claim and it's implications (merch has something already).

jagracar committed:
Adds the claimTokens code

jagracar committed:
Forgot to add new file

jagracar committed:
uncomment line again

jagracar commented on src/components/header/main_menu/MainMenu.jsx:
```diff
@@ -63,6 +63,13 @@ export const MainMenu = () => {
+            className={styles.menu_label}
+            label="Claim DAO tokens"
+            route="dao"
+            need_sync={!currentName || !currentAddress}
```
Not sure if this line is correct. What do you think @melMass?

jagracar commented on src/context/userStore.ts:
```diff
@@ -403,6 +409,30 @@ export const useUserStore = create<UserState>()(
+          step('Claiming Teia DAO tokens', 'Claiming Teia DAO tokens', true)
+
+          // Download the distribution file mapping file
+          const mapping = await downloadFileFromIpfs(DISTRIBUTION_MAPPING_IPFS_PATH)
```
Doesn't handle possible download errors...

jagracar replied:
Fixed that in my last commit

jagracar commented on src/context/userStore.ts:
```diff
@@ -403,6 +409,30 @@ export const useUserStore = create<UserState>()(
+          const mapping = await downloadFileFromIpfs(DISTRIBUTION_MAPPING_IPFS_PATH)
+
+          // Download the Merkle data
+          const merkleDataPath = MERKLE_DATA_IPFS_PATHS[mapping[user_address]]
```
Doesn't handle the possibility that the user_address is not in the mapping dictionary. That could happen with new users

jagracar replied:
Fixed that in my last commits

jagracar commented on src/context/userStore.ts:
```diff
@@ -403,6 +409,30 @@ export const useUserStore = create<UserState>()(
+          console.log(merkleData)
+          try {
+            const dropContract = await Tezos.wallet.at(DAO_TOKEN_CLAIM_CONTRACT)
+            const batch = dropContract.methods.claim(merkleData.proof, merkleData.leafDataPacked)
```
This fails at the moment because the current claim contract was deployed to the testnet and not to the mainnet

jagracar commented on src/pages/dao/index.jsx:
```diff
@@ -0,0 +1,30 @@
+      </div>
+
+      <div className={styles.dao__outer__container}>
+        <p>Warning, warning, warning!</p>
```
We should fill this with all the claim info. @merchantcoppola has written already something

jagracar commented:
Hi @melMass and @xat This already should be working once the token claim contract is deployed to the mainnet. Could you check that i'm not doing something stupid? I put the code where i thought it fit best, but i'm not totally sure.
There are still many things to improve, but the general structure should be there. Feel free to change things.

jagracar committed:
Improves user feedback

jagracar commented:
This is how the claim page looks at the moment

![claim](https://github.com/teia-community/teia-ui/assets/5071535/1833b5ef-a854-4015-b565-4dc03e756383)

One option would be to remove/comment the direct link to the page in the menu and merge it. We will be able to test the page on production under teia.art/dao, but normal users will not see it

merchantcoppola committed:
add copytext

merchantcoppola committed:
a lot of fixes and removing junk at th4e top

merchantcoppola committed:
adding the imports back in

merchantcoppola committed:
small syntax fix

merchantcoppola committed:
styling adjustments, cleaning up

jagracar committed:
Merge pull request #1 from jagracar/merchantcoppola-patch-1

melMass commented:
Nice I will be able to look into it tomorrow if still needed!

jagracar committed:
Small edits

jagracar committed:
Improves css

jagracar committed:
Completes link and max token number per user

jagracar commented:
> Nice I will be able to look into it tomorrow if still needed!

Yes, please!

merchantcoppola committed:
Updating texts

adding numbers to the lists
refining info
shortening a bit

merchantcoppola committed:
further updates, shortening

jagracar committed:
Updates ipfs paths to the last and probably final distribution

jagracar committed:
Temporally remove direct link to the DAO claim page in menu

merchantcoppola committed:
Merge pull request #2 from jagracar/merchantcoppola-patch-2

Updating texts

jagracar committed:
Small edits

merchantcoppola committed:
small adjustments in copytext

Zir0h committed:
Fix typo

merchantcoppola committed:
Discord feedback

- Plural for all mentions of TEIA tokens
- moved button to bottom, made it the finals section so users have to go over it before clicking the button
- changed the instructions so teh bold parts are the actual start of the senbtences


missing atm:
- make button bigger and colorful
- CBB (optional): add a checkbox "i have read and understood the disclaimer) that is a condition for the button to cliuckable

jagracar committed:
Merge pull request #3 from jagracar/merchantcoppola-patch-3

melMass commented:
Will review today :) 
From a quick look it looks all nice but I would just use markdown for the body like we do for the about: https://github.com/teia-community/teia-ui/blob/main/src/pages/about/index.jsx which is sourcing this md: https://github.com/teia-community/teia-ui/blob/main/src/lang/en/about.md

jagracar committed:
Adds checkbox

jagracar commented:
> Will review today :) From a quick look it looks all nice but I would just use markdown for the body like we do for the about: https://github.com/teia-community/teia-ui/blob/main/src/pages/about/index.jsx which is sourcing this md: https://github.com/teia-community/teia-ui/blob/main/src/lang/en/about.md

Oh, i used the FAQ as reference :(

I guess we can put the introduction and the legal disclaimer inside 2 md files. But the page has some interactivity, via the checkbox.

melMass committed:
chore 🧹: adding the md file I mentioned

We can always revert it if it's causing issues!

melMass commented:
> I guess we can put the introduction and the legal disclaimer inside 2 md files. But the page has some interactivity, via the checkbox.

Yep agreed if there is a need for using markup in places splitting it makes sense. Is there anything else besides the claim button? I'm not sure the staging link is getting auto update so I'm seing only that for now but saw the checkbox mention :)

melMass commented on src/utils/ipfs.js:
@Zir0h would it make sense to use something else than ipfs.io?

@jagracar I think we can use this method to avoid hardcoding it: https://github.com/teia-community/teia-ui/blob/e18769a602c9210af12859e17db14bb8152dbf90/src/utils/index.ts#L40-L70

jagracar replied:
Not sure to which file do you refer to? The code in utils/ipfs.js ? I realized that that method is not very good. It assumes that the file that it downloads from ipfs is a json file....
it probably should be renamed to downloadJsonFileFromIpfs

I also don't know if there is a better place for that method.

melMass replied:
In `src/utils/index.js` you have a `CIDtoURL` method that will use the env `VITE_IPFS_DEFAULT_GATEWAY` to resolve a gateway. this is just to avoid hard coding it but not a big deal at all

Zir0h replied:
yeah I would prefer using `VITE_IPFS_DEFAULT_GATEWAY`

jagracar replied:
Done in my last commit

melMass replied:
I will be able to dedicate a bit of time tomorrow after lunch!

melMass commented on src/pages/dao/index.jsx:
@jagracar follow up on the PR comments, if we go the md route it seems like interactivity is only at the very end so no need to split imo

jagracar replied:
Yep. Before there was a button in the middle, but not it's all at the end

jagracar replied:
Oh, you already added the md files! Thanks!

melMass reviewed (changes requested):
Looks very good to me! Awesome work everyone.
My only "complaint" is using markdown but besides that it all looks good 🥇

jagracar commented:
> > I guess we can put the introduction and the legal disclaimer inside 2 md files. But the page has some interactivity, via the checkbox.
> 
> Yep agreed if there is a need for using markup in places splitting it makes sense. Is there anything else besides the claim button? I'm not sure the staging link is getting auto update so I'm seing only that for now but saw the checkbox mention :)

I will not touch the code for the next hours to give you time for the review. I will try to add the md files after that.

melMass committed:
chore 🧹: update md file

I was using the outdated staged version

melMass committed:
chore 🧹: use h1 for the main title

melMass committed:
style 💎: apply hooks

jagracar committed:
Use CIDToURL method

Zir0h committed:
Fix typo

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

jagracar committed:
Adds link to the claim page in the menu

jagracar committed:
Update contract address to the test claim contract in mainnet

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

melMass committed:
docs 📚: fix typo

formatting needs to be reapplied but let's do it before merge

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

melMass commented:
@merchantcoppola you can keep doing the fixes for text in this branch, I'll update it in #313 since we have a live preview here: https://teia-community.github.io/teia-ui/previews/pr-313/ and I will push the extra changes from #313 here before merging

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

melMass committed:
Merge branch 'main' into feature/add-token-claim-page

melMass commented:
@jagracar can you change your remote to here: https://github.com/teia-community/teia-ui/tree/feature/add-token-claim-page ? (see #318)

merchantcoppola committed:
Copytext update (#319)

* Copytext update

- removing fact sheet link no 2
- updating the guide with the new checkbox approach

merchantcoppola committed:
small text edits

melMass commented:
@jagracar Can you check that the hooks are reworking now? I fixed them since, because it makes it quite hard to diff stuff

https://github.com/teia-community/teia-ui/pull/312/commits/bea45a5e7e5108a43e113e8e7cdfec87bd0630d5

jagracar commented:
How do i check that? I merge what you have in master in teia-ui? And then try to build it? I

jagracar committed:
Merge branch 'teia-community:feature/add-token-claim-page' into feature/add-token-claim-page

melMass commented:
> How do i check that? I merge what you have in master in teia-ui? And then try to build it? I

You can change any arbitrary js ts jsx file, stage it and commit it, the hooks should run or fail, you can just uncommit the test afterward

melMass commented:
Just closing this one as it's getting a bit hard to follow

jagracar commented:
Tested it with this commit:
https://github.com/teia-community/teia-ui/pull/318/commits/03eaf003b81a2b9f772b4e457ac1a7245916c96f
